dao/src/main/java/se/fortnox/reactivewizard/db/MonoFluxSink.java

```diff
@@ -0,0 +1,92 @@
+package se.fortnox.reactivewizard.db;
+
+import reactor.core.Disposable;
+import reactor.core.publisher.FluxSink;
+import reactor.util.context.Context;
+import reactor.util.context.ContextView;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.LongConsumer;
+
+import static java.util.Objects.nonNull;
+
+public class MonoFluxSink<T> implements FluxSink<T> {
+    private final AtomicLong    numNextSignals;
+    private final FluxSink<T>   actual;
+    private final boolean       isMono;
+
+    private T firstElement;
+
+    public MonoFluxSink(FluxSink<T> actual, boolean isMono) {
+        this.actual = actual;
+        this.isMono = isMono;
+        this.numNextSignals = new AtomicLong(1);
+    }
+
+    @Override
+    public FluxSink<T> next(T t) {
+        if (isMono) {
+            long numSignals = numNextSignals.getAndIncrement();
+            if (numSignals > 1) {
+                actual.error(new RuntimeException("Attempting to signal next multiple times on a Mono"));
```

Comment:
Maybe this message is a bit too low-level. We could add a short sentence about the most likely cause.